Chore: Revamped Package to standard package structure

EffanByte opened the pull request:
Used Unity's editor to rearrange all of the files into a standard layout given by [https://docs.unity3d.com/6000.1/Documentation/Manual/cus-layout.html](url).
Primary changes include

- Moved dependencies folder into separate Editor/ and Runtime/ folders
- Created a new package.json file for unity and moved npm package file to Editor/
- Edited CommandManager.cs and UnityBrowserBridge.cs files to only run when unity editor is the platform
- Downgraded Castle Core to version 4.4.1 (file name not changed, shall update in next commit) to be compatible with NSubstitiute's .NET Standard 2.0 requirement. Previous version of 5.1.1 was using .NET Standard 2.1

EffanByte committed:
Fix: Revamped Package to standard package structure

EffanByte committed:
fix: Path to PlayroomkitDevManagerEditor.css Changed

Since the package can now be pulled from git URL, the path has been updated to /Packages/package instead of Assets/Package.

EffanByte committed:
Added node_modules.meta to prevent creation of node_modules empty folder on opening project

EffanByte committed:
Removed .meta file from repository as well

EffanByte committed:
Feat: Created a partially working NPM Install Editor Window

The window is currently a menu item at the top. Functionalities that currently works are
- Refreshing to check node if node works
- Setting nodejs file path
- Going to the download page

Functionality Left to Implement:
- Checking for node_modules
- Running the install script

EffanByte committed:
Fix: File Path Change As a Git Package

Changed file path to be consistent with folder location of git packages

EffanByte committed:
Fix: Install Button Directory Changes

Trying to fix the currently directory for npm install.

EffanByte committed:
Fix:  a Few Compilation Errors resolved, moved node modules folder to PlayroomKit/Editor

EffanByte committed:
Fix: Builds Weren't Working as NPMManager was being used in runtime

EffanByte committed:
Fix: Fixed Vite config outputting .jslib to wrong directory

EffanByte committed:
Feat: Added Chrome Driver Download Button to NPM Manager

momintlh commented:
Need to move node_modules to Assets/playroom/editor, we will also use this for prefabs which come with the package

EffanByte committed:
Chore: Node Modules to Assets + UI Changes

- Automatically refreshes npm manager now after install
- Node Modules moved to Assets/Playroom from Packages/Playroom
